### PR TITLE
Add documentation about correct usage of `configure_project`

### LINKED ISSUE #3704: Create documentation of the usage of `configure_project`
## Description

`configure_project` should be used when multi-processing or multi-threading are used on Windows or Mac, we may not have the best generic way to solve this but we should put a section in the docs so at least there are something that people can start with.

### PR DESCRIPTION
**Pull request by noklam**
## Description

Fix #3704. This is tricky to resolve just with the error message. If users are using Kedro with `multiprocessing` themselves, they need to handle logging and `configure_project` carefully. There are no public API that we provide for user.

## Review Notes
The issue is discovered when we try to fix running `kedro-viz` with ParallelRunner. At the end we didn't fix the issue in kedro-viz, as it turns out to be quite tricky to implement hook that works with `ParallelRunner`. We don't have an working example so I didn't include anything in this PR.

The change adds explanation of the usage of `configure_project`, and try to give a pointer when people see this mysterious error message.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by astrojuanlu** [comment]
@noklam Do you intend to keep working on this? Otherwise we can close the PR and properly groom the issue at some other time

**Comment by noklam** [comment]
@astrojuanlu I fixed the PR and is ready to be reviewed.

**Review by astrojuanlu — APPROVED** [review]
These docs look good, thanks @noklam !

...However, is there a chance we open an issue about the underlying problem? Maybe there's something we can do at the API level to make this easier to get right the first time.

File: docs/source/kedro_project_setup/session.md [hunk 1]
```diff
@@ -63,3 +63,8 @@ This function uses `configure_project`, and additionally reads metadata from `py
 ### `configure_project`
 
 This function reads `settings.py` and `pipeline_registry.py` and registers the configuration before Kedro's run starts. If you have a packaged Kedro project, you only need to run `configure_project` before executing your pipeline.
+
+#### ValueError: Package name not found
+> ValueError: Package name not found. Make sure you have configured the project using 'bootstrap_project'. This should happen automatically if you are using Kedro command line interface.
+
+If you are using `multiprocessing`, you need to be careful about this. Depending on your Operating System, you may have [different default](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods). If the processes are `spawn`, Python will re-import all the modules in each process and thus you need to run `configure_project` again at the start of the new process. For example, this is how Kedro handle this in `ParallelRunner`(https://github.com/kedro-org/kedro/blob/9e883e6a0ba40e3db4497b234dcb3801258e8396/kedro/runner/parallel_runner.py#L84-L85)
```
Comment (merelcht) [inline docs/source/kedro_project_setup/session.md:70]:
```suggestion
If you are using `multiprocessing`, you need to be careful about this. Depending on your Operating System, you may have a [different default to start a process](https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods). If the processes are `spawn`, Python will re-import all the modules in each process and thus you need to run `configure_project` again at the start of the new process. For example, this is how Kedro handles this in `ParallelRunner`(https://github.com/kedro-org/kedro/blob/9e883e6a0ba40e3db4497b234dcb3801258e8396/kedro/runner/parallel_runner.py#L84-L85)
```

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Comment by noklam** [comment]
> ...However, is there a chance we open an issue about the underlying problem? Maybe there's something we can do at the API level to make this easier to get right the first time.


@astrojuanlu I don't know exactly what the issue will be, we have some issues and discussion opened that is related to this problem. https://github.com/kedro-org/kedro-viz/issues/1801#issuecomment-2035086525

The most concrete problem I had so far is "How to write a stateful hook that can run with ParallelRunner".

**Comment by astrojuanlu** [comment]
What I didn't fully understand is why `configure_project` or `bootstrap_project` break (or behave differently) under the `ParallelRunner`

**Pull request merged**